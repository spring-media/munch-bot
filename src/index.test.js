jest.mock('node-fetch')
const nfetch = require('node-fetch')
const fs = require('fs')

// const log = console.log
console.log = () => {}

describe('index', () => {
  it('should print the menu', async () => {
    jest.useFakeTimers()
    jest.setSystemTime(new Date(2019, 10, 15))

    process.env.SLACK_CHANNEL = 'channelID'
    process.env.SLACK_TOKEN = 'SlackToken'

    nfetch
      .mockResolvedValueOnce({ json: () => JSON.parse(fs.readFileSync('src/tests/example-2019-11-15.json', 'utf8')) })
      .mockResolvedValueOnce({ ok: true })

    const index = require('./index')
    await index.handler()
    const slackCall = nfetch.mock.calls[1]
    const slackPost = JSON.parse(slackCall[1].body)
    delete slackCall[1].body
    expect(slackCall).toMatchSnapshot()
    expect(slackPost).toMatchSnapshot()
  })
})
