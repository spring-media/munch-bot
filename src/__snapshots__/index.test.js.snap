// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should print the menu 1`] = `
Array [
  "https://hooks.slack.com/services/SlackToken",
  Object {
    "method": "POST",
  },
]
`;

exports[`index should print the menu 2`] = `
Object {
  "channel": "channelID",
  "text": "_The Munch-Bot kindly presents:_ *Das Menü von heute (13.8.2021)*


*------Papa-----*

_Essentia_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Spaghetti | Karotten-Pesto | Blattpetersilie>\` _6.90€ (*3.40€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Spaghetti Carbonara>\` _7.30€ (*3.80€*)_


_Tagesgericht_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Gebratenes Seelachsfilet | Kartoffelwürfel | Balsamico-Linsengemüse>\` _7.65€ (*4.15€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Vegane Kartoffelspalten | Ofengemüse | Curry-Dip>\` _7.10€ (*3.60€*)_


_Aktion_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Putenbrust in Kräuter Ei | Gebackene Kartoffeln | Feldsalat | Kräuteröl>\` _8.00€ (*4.50€*)_


_Vegetarisch_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Gebackener Feta | Mediterranes Gemüse | Rucola | Olivenöl>\` _7.00€ (*3.50€*)_


_Spezial_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Rindergulasch | Kartoffelpüree | Sauerrahm | Blattpetersilie>\` _7.30€ (*3.80€*)_


_Dessert_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Grießbrei | Früchtemix>\` _3.90€ (*1.75€*)_


_Suppe/Eintopf_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=papa|Kartoffelsuppe | Wurstscheiben>\` _3.90€ (*1.75€*)_


*------Canteen-----*

_Counter_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Schwarzer Quinoa | Süßkartoffel | Feldsalat>\` _8.30€ (*4.80€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Roastbeef | Frankfurter Grüne Sauce | Röstkartoffel>\` _9.20€ (*5.70€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Petersilienwurzel | Schwarzbrot | Thymian>\` _3.80€ (*1.71€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Cajun Spiced Lachs | Gemüse-Couscous | Limone | Ingwer>\` _8.50€ (*5.00€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Gefüllte Kürbis-Gnocchi | Baby Leaf | Tomate>\` _8.00€ (*4.50€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Indisches Dal | Granatapfel | Koriander>\` _7.00€ (*3.50€*)_


_Dessert_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Gelbe Grütze | Vanillesauce>\` _1.90€ (*0.85€*)_


_Salat_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Blattsalat | Balsamico | Croûtons>\` _6.00€ (*2.70€*)_


_ABEND_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Roastbeef | Frankfurter Grüne Sauce | Röstkartoffel>\` _9.20€ (*5.70€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Indisches Dal | Granatapfel | Koriander>\` _7.00€ (*3.50€*)_
 • \`<https://api.pace.berlin/foodfinder?lang=de&outlet=canteen|Currywurst | Fritten>\` _6.50€ (*3.00€*)_

_Preisangabe_: In Klammern mit Zuschuss von 3.50€
_Source_: <https://pace.berlin/|PACE>

*Do you have questions, comments or improvements? Feel free to ask questions here.*",
}
`;
